src/lib/pickles/pickles.ml

```diff
@@ -1504,6 +1538,82 @@ let%test_module "test no side-loaded" =
               Proof.verify_promise [ ((input, res), b0) ] ) ) ;
         ((input, res), b0)
     end
+
+    module Auxiliary_return = struct
+      module Statement = struct
+        type t = Field.t
+
+        let to_field_elements x = [| x |]
+
+        module Constant = struct
+          type t = Field.Constant.t [@@deriving bin_io]
+
+          let to_field_elements x = [| x |]
+        end
+      end
+
+      let tag, _, p, Provers.[ step ] =
+        Common.time "compile" (fun () ->
+            compile_promise
+              (module Statement)
```

Comment:
note: I think you said these could be removed, but I guess it would be for another PR